### ISSUE TITLE
Migrate to random.stateless_* api

### ISSUE THREAD

**Issue by LukeWood** [keras-team/keras-cv#258]
https://www.tensorflow.org/api_docs/python/tf/random/stateless_normal

**Comment by LukeWood** [comment]
Bad news - this is non trivial. seeds require 2 numbers for the stateless_uniform API. This leaves us with inconsistent random seed behavior - some layes require 1 int as a seed, some two.

**Comment by sebastian-sz** [comment]
@LukeWood regarding reproducible results across invocations - I'm wondering if `tf.random.Generator` class isn't good enough for this.
https://www.tensorflow.org/guide/random_numbers#the_tfrandomgenerator_class 

This would also be consistent with seed as a single integer. Although I'm not quite sure about the exact differences between stateless ops and Generator.

**Comment by LukeWood** [comment]
I think @qlzh727 has a lot of context on this issue.  you input here would be appreciated - thanks!

**Comment by qlzh727** [comment]
The BaseImageAugmentationLayer has a `_random_generator` which will handle this under the hood. User just need to call `_random_generator. random_normal/uniform()` etc.

**Comment by LukeWood** [comment]
How should we handle this for the `FactorSampler` class?  Should we export `_random_generator` under `__internal__ for now?

**Comment by qlzh727** [comment]
Humm, we might consider making the backend.RandomGenerator to be an internal API, but it might have some updates soon (I am fixing some stateless RNG bug), so the API interface might change. For now, we might want to put a TODO for using stateless RNG.

**Comment by LukeWood** [comment]
This is probably easier to just do when we move FactorSampler to core.  @fchollet and I discussed moving it to core keras, but after v1 so we can test out the API a bit more before moving to core Keras.  So this can remain a TODO.

**Comment by qlzh727** [comment]
I think we can leave this one out for the v0.1 release.

**Comment by bhack** [comment]
We need to check also other random operations that we have introduced in the library like `random_shuffle`. 
See more at https://github.com/keras-team/keras-cv/issues/259#issuecomment-1090425529

**Comment by bhack** [comment]
Just a reminder:
Currently all the currenty `tf.vectorized_map` convereters for `random.stateless_*` are going to fallback:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/parallel_for/pfor.py#L3372-L3389

**Comment by LukeWood** [comment]
This is primarily handled by `BaseImageAugmentationLayer`, so we can close for now.
